src/main/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/services/GoalService.kt

```diff
@@ -243,10 +244,12 @@ class GoalService(
     // also need to set the new targetDate
     if (updatedGoal.targetDate != null) {
       goalEntity.targetDate = LocalDate.parse(updatedGoal.targetDate)
+      goalEntity.reminderDate = null
```

Comment:
I think we should add this line to the achieveGoal function as well, since it's possible to mark a future goal as achieved and I presume we don't want to have reminders set for achieved goals.

Reply:
Agreed, good spot. It would also make sense for us to set it to null in removeGoal too, for the same reason. If a goal is re-added to the plan then the reminder can be set there.